src/traversals/greedy_color.jl

```diff
@@ -0,0 +1,155 @@
+
+     
+        for i in one(T):nvg
+            if colors_used[i] == false
+                cols[v] = i
+                break;
+            end
+        end
+    end
+
+    result = coloring{T}(maximum(cols), cols)
+    return result
+end
+
+"""
+    degree_greedy_color(g)
+
+Color graph `g` iteratively in the descending order of the degree of the vertices.
+"""
+function degree_greedy_color(g::AbstractGraph{U}) where U<:Integer
+    seq::Vector{U} = sortperm(degree(g, vertices(g)) , rev=true)
+    result = perm_greedy_color(g, seq)
+    return result
+end
+
+"""
+    parallel_random_greedy_color(g, reps)
+
+Color graph `g` iteratively in a random order using a greedy heuristic and
+choose the best coloring out of `reps` number of colorings computed in parallel.
+"""
+function parallel_random_greedy_color(
+    g::AbstractGraph,
+    reps::Integer
+    ) 
+
+    best::coloring = @parallel (best_color) for i in 1:reps
+        seq = shuffle(vertices(g))
+        perm_greedy_color(g, seq)
+    end
+
+    return best
+end
+
+"""
+    seq_random_greedy_color(g, reps)
+
+Color graph `g` iteratively in a random order using a greedy heuristic
+and choose the best coloring out of `reps` such random coloring.
+"""
+function seq_random_greedy_color(
+    g::AbstractGraph, 
+    reps::Integer
+    )
+
+    seq = shuffle(vertices(g))
+    best::coloring = perm_greedy_color(g, seq)
+
+    for i in 2:reps
+        shuffle!(seq)
+        best = best_color(best, perm_greedy_color(g, seq))
+    end
+    return best
+end
+
+"""
+    random_greedy_color(g, reps=1, parallel=false)
+
+Color graph `g` iteratively in a random order using a greedy heruistic
+and choose the best coloring out of `reps` such random coloring.
+
+If parallel is true then the colorings are executed in parallel.
+"""
+function random_greedy_color(
+    g::AbstractGraph, 
+    reps::Integer = 1,
+    parallel::Bool = false
+    ) 
+
+    if !parallel
+       return seq_random_greedy_color(g, reps)
+   else
+        return parallel_random_greedy_color(g, reps)
+    end
+
+end
+
+"""
+    greedy_color(g; sort_degree=false, parallel=false, reps = 1)
+
+Color graph `g` based on [Greedy Coloring Heuristics](https://en.wikipedia.org/wiki/Greedy_coloring)
+
+The heuristics can be described as choosing a permutation of the vertices and assigning the 
+lowest color index available iteratively in that order.
+
+If `sort_degree` is true then the permutation is chosen in reverse sorted order of the degree of the vertices.
+`parallel` and `reps` are irrelevant in this case.
+
+If `sort_degree` is false then `reps` colorings are obtained based on random permutations and the one using least
+colors is chosen.
+
+If `parallel` is true then this function executes coloring in parallel.
+"""
+function greedy_color(
+    g::AbstractGraph{U};
+    sort_degree::Bool=false,
+    parallel::Bool = false,
+    reps::Integer=1,
+)::coloring{U}   where U <: Integer
+
+    result = coloring(zero(U), zeros(U, nv(g)))
```

Comment:
why are you assigning `result` to a value here when you're overwriting it below?